### PR TITLE
feat: Add patch domain

### PR DESCRIPTION
**Pull request by bukinoshita**

### CODE REVIEW

File: src/domains/domains.spec.ts [hunk 1]
```diff
@@ -498,6 +499,41 @@ describe('Domains', () => {
     });
   });
 
+  describe('patch', () => {
+    it('oatch domain click tracking', async () => {
```
Comment (zenorocha) [inline src/domains/domains.spec.ts]: oatch > patch

Comment (bukinoshita) [inline src/domains/domains.spec.ts]:
```suggestion
    it('patch domain click tracking', async () => {
```

File: src/domains/domains.ts [hunk 2]
```diff
@@ -49,6 +54,17 @@ export class Domains {
     return data;
   }
 
+  async patch(payload: PatchDomainsOptions): Promise<PatchDomainsResponse> {
+    const data = await this.resend.patch<PatchDomainsResponseSuccess>(
+      `/domains/${payload.id}`,
+      {
+        click_tracking: payload.click_tracking,
+        domain_tracking: payload.domain_tracking,
```
Comment (zenorocha) [inline src/domains/domains.ts]: `domain_tracking`? is this supposed to be `open_tracking` instead?

Comment (bukinoshita) [inline src/domains/domains.ts]:
```suggestion
        open_tracking: payload.open_tracking,
```

File: src/domains/interfaces/patch-domain.interface.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { ErrorResponse } from '../../interfaces';
+import { Domain } from './domain';
+
+export interface PatchDomainsOptions {
+  id: string;
+  click_tracking?: boolean;
+  domain_tracking?: boolean;
```
Comment (bukinoshita) [inline src/domains/interfaces/patch-domain.interface.ts]:
```suggestion
  open_tracking?: boolean;
```

**Review by zenorocha — APPROVED** [review]

**Pull request merged**